crates/yulgen/tests/snapshots/yulgen__abi_decode_data_u256_bytes_string_bool_address_bytes_calldata_function.snap

```diff
@@ -0,0 +1,31 @@
+---
+source: crates/yulgen/tests/yulgen.rs
+expression: "abi_functions::decode_data(&[FixedSize::u256(),\n                             FixedSize::Array(Array{inner:\n                                                        Base::Numeric(Integer::U8),\n                                                    size: 100,}),\n                             FixedSize::String(FeString{max_size: 42,}),\n                             FixedSize::bool(), FixedSize::address(),\n                             FixedSize::Array(Array{inner:\n                                                        Base::Numeric(Integer::U8),\n                                                    size: 100,})],\n                           AbiDecodeLocation::Calldata)"
+
+---
+function abi_decode_data_uint256_bytes_100_string_42_bool_address_bytes_100_calldata(head_start, data_end) -> return_val_0, return_val_1, return_val_2, return_val_3, return_val_4, return_val_5 {
+    let encoding_size := sub(data_end, head_start)
+    if or(lt(encoding_size, 544), gt(encoding_size, 608)) { revert(0, 0) }
+    let head_offset_0 := 0
+    let head_offset_1 := 32
+    let head_offset_2 := 64
+    let head_offset_3 := 96
+    let head_offset_4 := 128
+    let head_offset_5 := 160
+    let decoded_val_0 := abi_decode_component_uint256_calldata(head_start, head_offset_0)
+    let decoded_val_1, data_start_offset_1, data_end_offset_1 := abi_decode_component_bytes_100_calldata(head_start, head_offset_1)
+    let decoded_val_2, data_start_offset_2, data_end_offset_2 := abi_decode_component_string_42_calldata(head_start, head_offset_2)
+    let decoded_val_3 := abi_decode_component_bool_calldata(head_start, head_offset_3)
+    let decoded_val_4 := abi_decode_component_address_calldata(head_start, head_offset_4)
+    let decoded_val_5, data_start_offset_5, data_end_offset_5 := abi_decode_component_bytes_100_calldata(head_start, head_offset_5)
+    if iszero(eq(data_start_offset_1, 192)) { revert(0, 0) }
+    if iszero(eq(data_start_offset_2, data_end_offset_1)) { revert(0, 0) }
+    if iszero(eq(data_start_offset_5, data_end_offset_2)) { revert(0, 0) }
+    if iszero(eq(encoding_size, data_end_offset_5)) { revert(0, 0) }
```

Comment:
@cburgdorf the Solidity code I looked at didn't revert with any specific values, but it looks like they plan on adding it. I'll create an issue to track this.

Reply:
Oh I didn't notice until know that you had a comment on this. Anyway, `0x99` is good for now :+1: